governance/program/src/instruction.rs

```diff
@@ -241,26 +243,45 @@ pub enum GovernanceInstruction {
     ///  By doing so you indicate you approve or disapprove of running the Proposal set of instructions
     ///  If you tip the consensus then the instructions can begin to be run after their hold up time
     ///
-    ///   0. `[writable]` Proposal account
-    ///   1. `[writable]` Token Owner Record account. PDA seeds: ['governance',realm, governing_token_mint, governing_token_owner]
-    ///   2. `[writable]` Proposal Vote Record account. PDA seeds: ['governance',proposal,governing_token_owner]  
-    ///   3. `[signer]` Governance Authority account
-    ///   4. `[]` Governance account
-    Vote {
+    ///   0. `[]` Governance account
+    ///   1. `[writable]` Proposal account
+    ///   2. `[writable]` Token Owner Record account. PDA seeds: ['governance',realm, governing_token_mint, governing_token_owner]
+    ///   3. `[signer]` Governance Authority (Token Owner or Governance Delegate)
+    ///   4. `[writable]` Proposal VoteRecord account. PDA seeds: ['governance',proposal,governing_token_owner_record]  
+    ///   5. `[]` Governing Token Mint
+    ///   6. `[signer]` Payer
+    ///   7. `[]` System program
+    ///   8. `[]` Rent sysvar
+    ///   9. `[]` Clock sysvar    
+    CastVote {
         #[allow(dead_code)]
         /// Yes/No vote
         vote: Vote,
     },
 
+    /// Finalizes vote in case the Vote was not automatically tipped within max_voting_time period
+    ///
+    ///   0. `[]` Governance account
+    ///   1. `[writable]` Proposal account    
+    ///   2. `[]` Governing Token Mint    
+    ///   3. `[]` Clock sysvar        
+    FinalizeVote {},
+
     ///  Relinquish Vote removes voter weight from a Proposal and removes it from voter's active votes
     ///  If the Proposal is still being voted on then the voter's weight won't count towards the vote outcome
     ///  If the Proposal is already in decided state then the instruction has no impact on the Proposal
     ///  and only allows voters to prune their outstanding votes in case they wanted to withdraw Governing tokens from the Realm
     ///
-    ///   0. `[writable]` Proposal account
-    ///   1. `[writable]` Token Owner Record account. PDA seeds: ['governance',realm, governing_token_mint, governing_token_owner]
-    ///   2. `[writable]` Proposal Vote Record account. PDA seeds: ['governance',proposal,governing_token_owner]
-    ///   3. `[signer]` Governance Authority account
+    ///   0. `[]` Governance account    
+    ///   1. `[writable]` Proposal account
+    ///   2. `[writable]` TokenOwnerRecord account. PDA seeds: ['governance',realm, governing_token_mint, governing_token_owner]
+    ///   3. `[writable]` Proposal VoteRecord account. PDA seeds: ['governance',proposal,governing_token_owner_record]
+    ///   4. `[]` Governing Token Mint    
+
+    ///   5. `[signer]` Optional Governance Authority (Token Owner or Governance Delegate)
+    ///       It's required only when Proposal is still being voted on    
```

Comment:
Yeah, I was thinking about that or maybe in the UI once `Proposal` is completed (decided or cancelled) there would be an option to prune it for yourself and others. However I'm not sure how expensive it's going to be if you have a large voting population so I decided to keep it optional.
Also you don't really have to prune your votes at all unless you want to withdraw your governing tokens from a `Realm`. And I was thinking it could be done in a simpler  and cheaper way without pruning all your past votes. 
All we need is to know how many pending Proposals are at the moment in the Realm and then you have to submit an instruction (let's say `ReleaseVoteStakes`) with the pending Proposals only and we can reset you staked_votes_count to 0 if you don't participate in them.

Reply:
Is it valid for someone to want to recover their lamports from past votes?  If we have 100 votes, that could be a good amount of lamports that I want to recover from my old voting record.

Reply:
No, we can't recover that because we want to know how users have voted historically  and `VoteRecord` account holds that information and hence can't be disposed.
When we `Relinquish` a vote which is already completed we just flag it for the owner so we know there are no outstanding pending votes but we do not destroy the `VoteRecord`